controllers/grid/publications/ExportPublishedPublicationsListGridHandler.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+/**
+ * @file controllers/grid/publications/ExportPublishedPublicationsListGridHandler.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ExportPublishedPublicationsListGridHandler
+ *
+ * @ingroup controllers_grid_publications
+ *
+ * @brief Handle exportable published publications list grid requests.
+ */
+
+namespace APP\controllers\grid\publications;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\issue\Collector;
+use APP\plugins\PubObjectsExportPlugin;
+use PKP\controllers\grid\feature\PagingFeature;
+use PKP\controllers\grid\feature\selectableItems\SelectableItemsFeature;
+use PKP\controllers\grid\GridColumn;
+use PKP\controllers\grid\GridHandler;
+use PKP\plugins\PluginRegistry;
+use PKP\security\authorization\PolicySet;
+use PKP\security\authorization\RoleBasedHandlerOperationPolicy;
+use PKP\security\Role;
+
+class ExportPublishedPublicationsListGridHandler extends GridHandler
+{
+    public PubObjectsExportPlugin $_plugin;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->addRoleAssignment(
+            [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+            ['fetchGrid', 'fetchRow']
+        );
+    }
+
+    //
+    // Implement template methods from PKPHandler
+    //
+    /**
+     * @copydoc PKPHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        $rolePolicy = new PolicySet(PolicySet::COMBINING_PERMIT_OVERRIDES);
+
+        foreach ($roleAssignments as $role => $operations) {
+            $rolePolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, $role, $operations));
+        }
+        $this->addPolicy($rolePolicy);
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc GridHandler::initialize()
+     *
+     * @param null|mixed $args
+     */
+    public function initialize($request, $args = null)
+    {
+        parent::initialize($request, $args);
+        $context = $request->getContext();
+
+        // Basic grid configuration.
+        $this->setTitle('plugins.importexport.common.export.publications');
+
+        $pluginCategory = $request->getUserVar('category');
+        $pluginPathName = $request->getUserVar('plugin');
+
+        // Allow for plugins that have already been loaded (e.g. by injection from a generic plugin)
+        // Otherwise, load the plugin as specified
+        $this->_plugin = PluginRegistry::getPlugin($pluginCategory, $pluginPathName) ?? PluginRegistry::loadPlugin($pluginCategory, $pluginPathName);
```

Comment:
When using the search feature in the DOAJ plugin, an error happens around here and it doesn't seem to load the plugin (maybe due to the importexport category vs the plugin location?): 
```
PHP Fatal error:  Uncaught TypeError: Cannot assign null to property APP\controllers\grid\publications\ExportPublishedPublicationsListGridHandler::$_plugin of type APP\plugins\PubObjectsExportPlugin in /Users/kaitlinnewson/Code/ojs-main/controllers/grid/publications/ExportPublishedPublicationsListGridHandler.php:84
```

Reply:
Oh, yes, moving it to the generic plugin changes a few things...